docs/src/index.md

```diff
@@ -1,3 +1,218 @@
 # Methode_of_conjugate_gradients.jl
 
-Documentation for Methode_of_conjugate_gradients.jl
+This is the documentation for the repository of the method of Conjugate Gradients.jl. The documentation is divided into two parts. The first part provides a mathematical explanation of Conjugate Gradients and Gradient Descent. The second part provides a detailed explanation of the code.
+
+## The mathematical explanation
+### Graph Embedding using Conjugate Gradient Method
+
+In the context of graph embedding, the conjugate gradient method can be applied to find an equilibrium position for the nodes in a plane or space using a physical method. The idea is to treat the graph as a system of masses connected by springs, where the goal is to find the equilibrium positions of the nodes that minimize the total potential energy of the system. The potential energy function can be defined based on the distance between neighboring nodes.
+
+The equation given for each coordinate (xi, yi, zi) of the vertices of the graph in space is:
+
+$$
+\begin{aligned}
+    -st(i) x_i + \sum_{j \in N(i)} x_j - st(i) y_i + \sum_{j \in N(i)} y_j &= 0, \\
+    -st(i) y_i + \sum_{j \in N(i)} y_j - st(i) z_i + \sum_{j \in N(i)} z_j &= 0, \\
+    -st(i) z_i + \sum_{j \in N(i)} z_j - st(i) x_i + \sum_{j \in N(i)} x_j &= 0.
+\end{aligned}
+$$
+
+Here, st(i) represents the stage of the i-th node, and N(i) is the set of indices of neighboring nodes. This equation ensures that the total force acting on each node is zero in equilibrium. If some nodes are fixed, the others will occupy an equilibrium position between the fixed nodes.
+
+To solve this system of equations using the conjugate gradient method, we can first rewrite the equations in matrix form Ax = b, where A is a sparse matrix representing the graph structure, and x and b are vectors containing the coordinates of the vertices and the right-hand side of the equation, respectively. The conjugate gradient method can then be used to solve this linear system iteratively, finding the equilibrium positions of the nodes in the graph.
+
+The example provided in the code section demonstrates how to create a sparse matrix representation of the graph embedding problem, and then use the conj_grad function to solve it. 
+
+### Conjugate gradient method
+The conjugate gradient method is an iterative algorithm used to solve linear systems of equations, particularly for symmetric and positive-definite matrices. It converges faster than other iterative methods like gradient descent, especially when dealing with large and sparse matrices.
+### The Gradient descent method
+Gradient descent is an optimization algorithm that tries to minimize a given objective function by iteratively moving in the direction of the steepest descent, as defined by the negative of the gradient. In this code, gradient descent is applied to a 2D quadratic function for visualization purposes.
+## The code explaind
```

Comment:
The explanation of code is in my opinion better done with comments inside the code itself.